nixos/tests/nixos-rebuild-install-bootloader.nix

```diff
@@ -60,7 +60,7 @@ import ./make-test-python.nix ({ pkgs, ... }: {
       # Need to run `nixos-rebuild` twice because the first run will install
       # GRUB anyway
       with subtest("Switch system again and install bootloader"):
-          result = machine.succeed("nixos-rebuild switch --install-bootloader")
+          result = machine.succeed("nixos-rebuild switch --install-bootloader 2>&1")
```

Comment:
Not sure if the `switch-to-configuration` messages going to the stderr instead of stdout count as a breaking change.

Reply:
A breaking fix? `stdout` isn't for messages - that's be quite non-standard.

Reply:
Wouldn't some messages go to stderr anyway?

Reply:
Actually, I am looking at the `install-grub.pl` code and it seems to send those messages to stderr anyway.

Not sure why it was working before and not it isn't (the asserts were not working after the `--pipe --scope` change).

Reply:
Here are the messages:
- https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/system/boot/loader/grub/install-grub.pl#L99
- https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/system/boot/loader/grub/install-grub.pl#L750

Reply:
> why it was working before

A pty is a single file descriptor iirc, so the information of which std stream was used was lost, with systemd-run deciding to send it to `stdout` apparently?

This make `--pipe` seem like a _fix_ more than anything.

Slightly unfortunate that we don't have a pty anymore, because we do have one in the Nix sandbox for instance (but with stdin from `/dev/null`). Not a big deal; it least it will consistently not have one from now on.